lib/beacon-verification.ts

```diff
@@ -66,13 +71,13 @@ export async function verifySigOnG1(
     domainSeparationTag= 'BLS_SIG_BLS12381G2_XMD:SHA-256_SSWU_RO_NUL_'
 ): Promise<boolean> {
     const P = normP2(publicKey);
-    const Hm = await normP1Hash(message, domainSeparationTag);
-    const G = PointG2.BASE;
+    const Hm = normP1Hash(message, domainSeparationTag);
+    const G = bls.G2.ProjectivePoint.BASE;
     const S = normP1(signature);
-    const ePHm = pairing(Hm, P.negate(), false);
-    const eGS = pairing(S, G, false);
-    const exp = eGS.multiply(ePHm).finalExponentiate();
-    return exp.equals(Fp12.ONE);
+    const ePHm = bls.pairing(Hm, P.negate(), false);
+    const eGS = bls.pairing(S, G, false);
+    const exp = bls.fields.Fp12.mul(eGS, ePHm);
```

Comment:
Is the final exponentation no longer necessary with updated noble curves?
From what I can see the DST looks correct

Reply:
It could be that. I am not sure how to compare GT elements though...